rules/integrations/github/initial_access_github_actions_workflow_injection_blocked.toml

```diff
@@ -0,0 +1,127 @@
+
+This rule detects attempts to push workflow files to a GitHub repository from within a GitHub Actions workflow that are blocked by GitHub's security controls. This is a key indicator of supply chain attacks where malicious code attempts to establish persistence by injecting backdoor workflows.
+
+### Possible investigation steps
+
+- Review the `github.repo` field to identify which repository was targeted.
+- Examine the `github.actor` to determine if the action was triggered by a bot (`github-actions[bot]`) or a user account (PAT-based).
+- Check recent workflow runs in the repository for suspicious activity, especially in jobs that run `npm install` or other package manager commands.
+- Review the repository's dependencies for recently added or updated packages that may contain malicious preinstall/postinstall hooks.
+- Examine the `github.reasons.message` field for details on which workflow file was being created or modified.
+- Search for other repositories in the organization that may have the same malicious dependency.
+- Review GitHub audit logs for successful workflow file modifications that may have occurred before protections were enabled.
+
+### False positive analysis
+
+- Legitimate automation tools that manage workflow files may trigger this alert. Verify if the repository uses tools like Dependabot, Renovate, or custom automation that modifies workflows.
+- CI/CD pipelines that intentionally update workflow files should use a PAT with the 'workflows' scope and be documented.
+
+### Response and remediation
+
+- If this is a confirmed attack attempt, immediately audit all dependencies in the affected repository.
+- Remove any suspicious packages and regenerate lock files.
+- Rotate any secrets that may have been exposed during the CI run.
+- Review and revoke any PATs that may have been compromised.
+- Enable branch protection rules requiring pull request reviews for workflow file changes.
+- Consider implementing CODEOWNERS for `.github/workflows/` directory.
+- Search for indicators of compromise such as unexpected workflow files (e.g., `discussion_*.yaml`, `formatter_*.yml`).
+"""
+references = ["https://www.wiz.io/blog/shai-hulud-2-0-ongoing-supply-chain-attack"]
+risk_score = 47
+rule_id = "e8b37f18-4804-4819-8602-4aba1169c9f4"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Use Case: Threat Detection",
+    "Tactic: Initial Access",
+    "Tactic: Persistence",
+    "Tactic: Execution",
+    "Data Source: Github",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-github.audit-* metadata _id, _index, _version
+| where
+    data_stream.dataset == "github.audit" and
+    event.action == "protected_branch.rejected_ref_update" and
+    github.category == "protected_branch" and
+    github.reasons.code == "workflow_updates" and
+    match(github.reasons.message::STRING, "refusing to allow a GitHub App to create or update workflow")
+| keep *
```

Comment:
As a note too, we probably need to update https://github.com/elastic/detection-rules/pull/5441 to handle this as well.

Reply:
No its not a blocker, this can go in without the test. Just a note that if we are fine with `keep *` then we need to update the schema validation to include this as a valid pattern.

Reply:
@Mikaayenson spoke with @eric-forte-elastic and explained why we are using a non-aggregation based ESQL query and that if we continue to enforce `keep` for all ESQL queries as informed early on upstream, then there may be ESQL queries where we need to `keep *` and should allow this. Plan is to merge this PR before #5441